tools/topology/sof/pipe-waves-codec-demux-playback.m4

```diff
@@ -0,0 +1,176 @@
+# Demux Volume Pipeline with Waves codec
+#
+#  Low Latency Playback with demux and volume.
+#
+# Pipeline Endpoints for connection are :-
+#
+#	Playback Demux
+#	B3 (DAI buffer)
+#
+#
+#  host PCM_P -- B0 --> Waves -> B1 --> volume -- B2 --> Demux(M) -- B3 --> sink DAI0
+#                                                              |
+#                                                          pipeline n+1 --> DAI
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`pga.m4')
+include(`muxdemux.m4')
+include(`mixercontrol.m4')
+include(`bytecontrol.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`codec_adapter.m4')
+
+define(`SETUP_PARAMS_NAME', `Waves' `ENDPOINT_NAME' `Setup' PIPELINE_ID)
+
+CONTROLBYTES_PRIV(PP_SETUP_CONFIG,
+`       bytes "0x53,0x4f,0x46,0x00,'
+`       0x00,0x00,0x00,0x00,'
+`       0x20,0x00,0x00,0x00,'
+`       0x00,0x10,0x00,0x03,'
+`       0x00,0x00,0x00,0x00,'
+`       0x00,0x00,0x00,0x00,'
+`       0x00,0x00,0x00,0x00,'
+`       0x00,0x00,0x00,0x00,'
+
+`       0x00,0x01,0x41,0x57,'
+`       0x00,0x00,0x00,0x00,'
+`       0x80,0xBB,0x00,0x00,'
+`       0x20,0x00,0x00,0x00,'
+`       0x02,0x00,0x00,0x00,'
+
+`       0x00,0x00,0x00,0x00,'
+`       0x0c,0x00,0x00,0x00,'
+
+`       0x00,0x00,0x00,0x00"'
+)
+
+# Post process Bytes control for setup config
+C_CONTROLBYTES(SETUP_PARAMS_NAME, PIPELINE_ID,
+        CONTROLBYTES_OPS(bytes),
+        CONTROLBYTES_EXTOPS(void, 258, 258),
+        , , ,
+        CONTROLBYTES_MAX(, 8192),
+        ,
+        PP_SETUP_CONFIG)
+
+define(`RUNTIME_PARAMS_NAME', `Waves' `ENDPOINT_NAME' `Runtime' PIPELINE_ID)
```

Comment:
I don't fully follow why here you have a PIPELINE ID but not in the other file without demux?

Reply:
`pipe-waves-codec-playback.m4` defines `CA_RUNTIME_CONTROLBYTES_NAME` and `CA_SETUP_CONTROLBYTES_NAME`
then `sof/pipe-codec-adapter-playback.m4` adds pipeline ID to it:
https://github.com/thesofproject/sof/blob/main/tools/topology/sof/pipe-codec-adapter-playback.m4#L73

Reply:
so here I am aligning with `sof/pipe-codec-adapter-playback.m4` implementation adding pipeline ID to name of a control